File: app/src/main/java/com/semisonfire/cloudgallery/ui/main/disk/adapter/items/GalleryItem.java
package com.semisonfire.cloudgallery.ui.main.disk.adapter.items;

import com.semisonfire.cloudgallery.data.model.Photo;

import java.util.ArrayList;
import java.util.List;

public class GalleryItem extends DiskItem {

    private List<Photo> photos;

    public GalleryItem() {
        photos = new ArrayList<>();
    }

    public List<Photo> getPhotos() {
        return photos;
    }

    public void setPhotos(List<Photo> photos) {
        this.photos = photos;
    }

    @Override
    public int getType() {
        return TYPE_GALLERY;
    }
}

File: app/src/main/java/com/semisonfire/cloudgallery/ui/disk/adapter/items/GalleryItem.kt
package com.semisonfire.cloudgallery.ui.disk.adapter.items

import com.semisonfire.cloudgallery.data.model.Photo
import java.util.ArrayList

class GalleryItem : DiskItem() {
    var photos: List<Photo> = ArrayList()

    fun getMutablePhotos(): MutableList<Photo> {
        return photos as MutableList<Photo>
    }

    override val type: Int
        get() = TYPE_GALLERY
}